src/app/frontend/deploy/createnamespace_controller.js

```diff
@@ -43,6 +43,18 @@ export default class NamespaceDialogController {
 
     /** @export {string} */
     this.namespace = '';
+
+    /**
+     * Max-length validation rule for namespace
+     * @export {number}
+     */
+    this.namespaceMaxLength = 63;
+
+    /**
+     * Pattern validation rule for namespace
+     * @export {string}
+     */
+    this.namespacePattern = "^[a-z0-9]([-a-z0-9]*[a-z0-9])?$";
```

Comment:
Can you have a RegEx instead of a string here? This would be faster and less error prone. Docs say that you can gave regex here: https://code.angularjs.org/1.4.7/docs/api/ng/directive/input